### DISCUSSION TITLE
Crazyflie Bolt 1.1 report ”SUP: Can not fly“

### DISCUSSION THREAD

**Discussion by justinleeyang** [bitcraze/discussions#1722] [General]
Hi，

Crazyflie Bolt 1.1 SYS: Self test passed, but report ”SUP: Can not fly“, why? 
```

SYS: Crazyflie Bolt 1.1 is up and running!
SYS: Build 32:9343aa686600 (2023.07 +32) CLEAN
SYS: I am 0x20323457594B500C002D0026 and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_INFO: Memory error: wrong header ID
DECK_INFO: Deck 0 has corrupt OW memory. No driver will be initialized!
DECK_CORE: 0 deck(s) found
IMU: BMI088: Using SPI interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Complementary (1) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushless motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2023.02 +1
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 20496 bytes
STAB: Starting stabilizer loop
SUP: Can not fly

```
![image](https://github.com/user-attachments/assets/a80d215f-e16c-4421-aea6-5a4a8b54e429)

**Comment by justinleeyang** [comment]
update newest firmware, the problem still exist:

```
SYS: Crazyflie Bolt 1.1 is up and running!
SYS: Build 0:b046745801fc (2024.10.2) CLEAN
SYS: I am 0x20373459424D5007003B001C and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_CORE: 0 deck(s) found
IMU: BMI088: Using SPI interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Complementary (1) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushless motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2024.10 (CB11)
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 17936 bytes
STAB: Starting stabilizer loop
SUP: Can not fly
```

**Reply by ToveRumar** [reply]
Hi!
This question is a bit hard to answer without more info. 
How does the flight tab look? 
Does the arm button show up? 

What software are you using, have you made any modifications? 
To be able to fly you need to have the bold on a stable and flat surface during startup. 

Could you tell me a bit more about what is happening and your application.

**Reply by justinleeyang** [reply]
> How does the flight tab look?

Flight tab can get right IMU data，When the board moves, the dashboard will also tilt:

> Does the arm button show up?

Didn't notice if this was displayed

> To be able to fly you need to have the bold on a stable and flat surface during startup.

Yes，my device hold on stable and Place horizontally on the table

> What software are you using, have you made any modifications?

1，nRF51822：
（1）mbs is modified to bolt info from crazyflie2.1, address is 3FFE0. and the file name is
nrf_mbs_cf21_bolt.hex, then copy to crazyflie2-nrf-firmware/bootloaders/
![image](https://github.com/user-attachments/assets/0018f5bd-0eb8-46b0-a0ee-fe32ef37a927)


（2）crazyflie2-nrf-firmware's Makefile add factory_reset_bolt, then execute the following command: 
`
make factory_reset_bolt PLATFORM=bolt
`

![image](https://github.com/user-attachments/assets/e283fc7d-ff99-44c6-bd1e-7380d32a55de)
![image](https://github.com/user-attachments/assets/2e6d2504-c404-4236-ab66-a69499a23ff3)

2，stm32f405：
```
make blot_defconfig
make 
```

**Reply by ToveRumar** [reply]
I would need to test this myself to give you a more precise answer, however I would try to arm it (by clicking the arm button in the cfclient). The bolt is unarmed by default and need to be armed to be able to take off. 

This might be the reason it never ends up in a "ready to fly "-state.

**Reply by justinleeyang** [reply]
Yes，Must Arm，then read to fly, Thanks

**Reply by ToveRumar** [reply]
Good to hear that we narrowed down your issue!
